### DISCUSSION TITLE
Is there an API for Tree Find Control?

### DISCUSSION THREAD

**Discussion by alefragnani** [microsoft/vscode-discussions#141] [Extension Development QnA] [answered]
I would like to mimic the behavior provided in this mockup https://github.com/microsoft/vscode/issues/146806#issuecomment-1122758195, where the team were discussing the discoverability of filtering in tree view. I would like to add a menu item that would call the Tree Find Control and display it to filter the tree view where the menu item is located. But unfortunately, I didn't find an API for that.

I could reproduce the basic behavior I want, but I wonder if there is a proper/more elegant way to do that.

```ts
    await vscode.commands.executeCommand("projectsExplorerGit.focus");
    await vscode.commands.executeCommand("list.find");
```

* The first line focuses the view where I want to filter
* The second line calls the Tree Find Control API

Am I missing something?

Thanks in advance

**Comment by isidorn** [comment, marked as answer]
Maybe @alexr00 can help

**Reply by alexr00** [reply]
What you have with the 2 commands is the only way to accomplish this at the moment.

**Reply by alefragnani** [reply]
Thank you @alexr00 !
